runtime/lua/vim/diagnostic.lua

```diff
@@ -890,22 +872,34 @@ M.handlers.signs = {
     end
 
     local ns = M.get_namespace(namespace)
-    if not ns.user_data.sign_group then
-      ns.user_data.sign_group = string.format('vim.diagnostic.%s', ns.name)
+    if not ns.user_data.sign_ns then
+      ns.user_data.sign_ns = api.nvim_create_namespace('')
     end
 
-    local sign_group = ns.user_data.sign_group
+    opts.signs.text = vim.tbl_extend(
+      'force',
+      vim
+        .iter({ unpack(M.severity, 1, 4) })
+        :map(function(v)
+          return v:sub(1, 1):upper()
+        end)
+        :totable(),
+      opts.signs.text or {}
+    )
+
     for _, diagnostic in ipairs(diagnostics) do
-      vim.fn.sign_place(0, sign_group, sign_highlight_map[diagnostic.severity], bufnr, {
+      local text = opts.signs.text[diagnostic.severity] or 'U'
```

Comment:
Along with my suggestion above to use a local `text` table instead of modifying `opts.signs.text`:

```suggestion
      local sign_text = text[diagnostic.severity] or 'U'
```